### PR TITLE
Added new template to deploy two-tier .net

### PR DESCRIPTION
**Pull request by wmhussain**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Two tier .Net app and MS SQL database 
*
*

### CODE REVIEW

File: two-tier-app-migration-to-containers-on-Azure/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+*	Fill in all the required parameter values. Accept the terms and condition on click Purchase.
+*	Access the deployment job once it is successful. In deployment job output, you will find the Public IP Address of VMs which can be used connect to the VMs.
+*	Open MS SQL Studio and verify the database is successfully created
+*	Open IIS Manager and verify website is created and launch the same in browser.
+*	Follow the post deployment configuration document [here](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/two-tier-app-migration-to-containers-on-Azure/Two-tier%20App%20migration%20on%20Azure%20Container.pdf) to learn about further configuration.
+## Deployment Time
+The deployment takes about 20 minutes to complete.
+## Support
+For any support related questions, issues or customization requirements, please contact info@spektrasystems.com <br/>
+****End****
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/README.md]: remove the PDF file from the sample

File: two-tier-app-migration-to-containers-on-Azure/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+        }
+    },
+    "variables": {
+        "storageName": "[concat('stg', uniqueString(resourceGroup().id))]",
+        "storageAccountType": "Standard_LRS",
+        "virtualNetworkName": "VNET",
+        "vnetAddressPrefix": "10.0.0.0/16",
+        "subnet1Name": "Frontend-subnet",
+        "subnet1Prefix": "10.0.0.0/24",
+        "subnet1Ref": "[concat(variables('vnetID'),'/subnets/', variables('subnet1Name'))]",
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/azuredeploy.json]: simplify and just use resourceId()

File: two-tier-app-migration-to-containers-on-Azure/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+        "publicIPAddressType": "Dynamic",
+        "publicIPAddressName": "PublicIP",
+        "dnsLabelPrefix":"[concat('dns', uniqueString(resourceGroup().id))]",
+        "networkSecurityGroupName": "Nsg",
+        "nicName": "NIC",
+        "vnetId": "[resourceId('Microsoft.Network/virtualNetworks', variables('virtualNetworkName'))]",
+        "subnetRef": "[concat(variables('vnetId'), '/subnets/', variables('subnet1Name'))]",
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/azuredeploy.json]: simplify and just use resourceId

File: two-tier-app-migration-to-containers-on-Azure/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+        "sqlAuthenticationPassword": "dbPassw0rd",
+        "rServicesEnabled": "false",
+        "scriptsFolder": "scripts/",
+        "nestedFolder": "nested/",
+        "twoTierAppTags": {
+        "type": "object",
+        "provider": "NA"
+      },
+       "quickstartTags": {
+        "type": "object",
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/azuredeploy.json]: variables don't need type declarations

File: two-tier-app-migration-to-containers-on-Azure/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+        "name": "two-tier-app"
+      }
+    },
+    "resources": [{
+            "name": "[parameters('virtualMachineName')]",
+            "type": "Microsoft.Compute/virtualMachines",
+            "apiVersion": "2016-04-30-preview",
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/azuredeploy.json]: update apiVersion

File: two-tier-app-migration-to-containers-on-Azure/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,369 @@
...
+                "protectedSettings": {
+                    "SQLAuthUpdateUserName": "[variables('sqlAuthenticationLogin')]",
+                    "SQLAuthUpdatePassword": "[variables('sqlAuthenticationPassword')]"
+                }
+            }
+        },
+        {
+            "name": "[variables('storageName')]",
+            "type": "Microsoft.Storage/storageAccounts",
+            "apiVersion": "2015-06-15",
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/azuredeploy.json]: update apiVersion

File: two-tier-app-migration-to-containers-on-Azure/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,369 @@
...
+                        },
+                        "privateIPAllocationMethod": "Dynamic",
+                        "publicIpAddress": {
+                            "id": "[resourceId('Microsoft.Network/publicIpAddresses', variables('publicIpAddressName'))]"
+                        }
+                    }
+                }]  }
+        },
+        {
+            "apiVersion": "2015-05-01-preview",
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/azuredeploy.json]: update apiVersion

File: two-tier-app-migration-to-containers-on-Azure/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,369 @@
...
+                }
+            }
+        }
+    ],
+    "outputs": {
+        "adminUsername": {
+            "type": "string",
+            "value": "[parameters('adminUsername')]"
+        },
+        "adminPassword":{
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/azuredeploy.json]: remove this

File: two-tier-app-migration-to-containers-on-Azure/nested/iis.json [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+      },
+      "defaultValue": ""
+    }
+  },
+  "variables": {
+    "vmExtensionName": "WebDeploy",
+    "nestedCustomTemplateFileName": "webDeploy.json",
+    "DSCFolder": "DSC/",
+    "nestedFolder": "nested/",
+    "dscExtensionFileName": "ContosoWebsite.ps1.zip",
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/nested/iis.json]: need to include the source ps1 file in the same - see BP guide re: DSC

File: two-tier-app-migration-to-containers-on-Azure/nested/iis.json [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+    "modulesUrl": "[concat(parameters('_artifactsLocation'), variables('DSCFolder'), variables('dscExtensionFileName'), parameters('_artifactsLocationSasToken'))]",
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/nested/iis.json]: use the uri() function to build the url

File: two-tier-app-migration-to-containers-on-Azure/nested/iis.json [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+   
+    "configurationFunction": "ContosoWebsite.ps1\\ContosoWebsite"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Compute/virtualMachines/extensions",
+      "name": "[concat(parameters('virtualMachineName'),'/', variables('vmExtensionName'))]",
+      "apiVersion": "2015-05-01-preview",
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/nested/iis.json]: update apiVersion

File: two-tier-app-migration-to-containers-on-Azure/nested/iis.json [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+      "name": "WebDeploy",
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2016-09-01",
+      "dependsOn": [
+        "[concat('Microsoft.Compute/virtualMachines/', parameters('virtualMachineName'),'/extensions/', variables('vmExtensionName'))]"
+      ],
+      "properties": {
+        "mode": "Incremental",
+        "templateLink": {
+          "uri": "[concat(parameters('_artifactsLocation'), variables('nestedFolder'), variables('nestedCustomTemplateFileName'), parameters('_artifactsLocationSasToken'))]",
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/nested/iis.json]: use the uri function

File: two-tier-app-migration-to-containers-on-Azure/nested/webDeploy.json [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+      "metadata": {
+        "description": "The sasToken required to access _artifactsLocation.  When the template is deployed using the accompanying scripts, a sasToken will be automatically generated."
+      },
+      "defaultValue": ""
+    }
+  },
+  "variables": {
+    "scriptFolder": "scripts/",
+    "scriptFileName": "envcreate.ps1",
+    "dbcreateScript": "[concat(parameters('_artifactsLocation'), variables('scriptFolder'), variables('scriptFileName'), parameters('_artifactsLocationSasToken'))]"
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/nested/webDeploy.json]: use uri function

File: two-tier-app-migration-to-containers-on-Azure/nested/webDeploy.json [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+  },
+  "resources": [
+    {
+      "apiVersion": "2015-06-15",
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/nested/webDeploy.json]: update apiVersion

File: two-tier-app-migration-to-containers-on-Azure/scripts/DotNetAppSqlDb.SetParameters.xml [hunk 15]
```diff
@@ -0,0 +1,5 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<parameters>
+  <setParameter name="IIS Web Application Name" value="crud" />
+  <setParameter name="MyDbConnection-Web.config Connection String" value="Data Source=localhost;Initial Catalog=cruddb;Integrated Security=False;User Id=dbuser;Password=dbPassw0rd;MultipleActiveResultSets=True" providerName="System.Data.SqlClient" />
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/scripts/DotNetAppSqlDb.SetParameters.xml]: don't hardcode the password

File: two-tier-app-migration-to-containers-on-Azure/scripts/DotNetAppSqlDb.deploy.cmd [hunk 16]
```diff
@@ -0,0 +1,311 @@
...
+echo In addition, you can change IIS Application Name, Physical path, 
+echo connectionString and other deploy parameters in the following file:
+echo "%_DeploySetParametersFile%"
+echo.
+echo For more information about this batch file, visit https://go.microsoft.com/fwlink/?LinkID=183544 
+) else (
+start notepad "%RootPath%DotNetAppSqlDb.deploy-readme.txt"
+)
+echo =========================================================
+goto :eof
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/scripts/DotNetAppSqlDb.deploy.cmd]: need source for DotNetAppSqlDb.zip
remove webdeploy msi from the sample and get that file from a secure location

File: two-tier-app-migration-to-containers-on-Azure/scripts/envcreate.ps1 [hunk 17]
```diff
@@ -0,0 +1,104 @@
+# to Install web deploy
+$Url = "https://raw.githubusercontent.com/wmhussain/two-tier-app-migration-containers/master/scripts/WebDeploy_amd64_en-US.msi"
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/scripts/envcreate.ps1]: pull this from is source location on ms.com

File: two-tier-app-migration-to-containers-on-Azure/scripts/envcreate.ps1 [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+
+#To change the default port of default website
+set-webbinding -Name 'Default Web Site' -BindingInformation "*:80:" -propertyName "Port" -Value "1234"
+
+#To create a crud website
+
+New-Website -Name crud -Force -PhysicalPath C:\inetpub\crud -Port 80
+
+#To download & deploy Crud app using local database
+$Url1 = "https://raw.githubusercontent.com/wmhussain/two-tier-app-migration-containers/master/scripts/DotNetAppSqlDb.deploy.cmd"
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/scripts/envcreate.ps1]: these url's must be passed into the script, not hardcoded

File: two-tier-app-migration-to-containers-on-Azure/scripts/envcreate.ps1 [hunk 19]
```diff
@@ -0,0 +1,104 @@
...
+$env:MONGODB_URL = "mongodb://localhost/meanstacktutorials"
+
+#Download Nodejs Application & Install
+$Url1 = "https://nodejs.org/dist/v8.8.1/node-v8.8.1-x64.msi"
+Invoke-WebRequest -Uri "$Url1" -OutFile "C:\Packages\node-v8.8.1-x64.msi"
+msiexec.exe /i C:\Packages\node-v8.8.1-x64.msi  /qn
+sleep 30
+
+#clone Nodejs ToDo app from Github
+$Url2 = "https://github.com/evillgenius75/gbb-todo/archive/master.zip"
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/scripts/envcreate.ps1]: this must be passed in

File: two-tier-app-migration-to-containers-on-Azure/scripts/DotNetAppSqlDb.deploy.cmd (hunk 16 above)
Comment (wmhussain) [reply two-tier-app-migration-to-containers-on-Azure/scripts/DotNetAppSqlDb.deploy.cmd]: removed webdeploy and provided link to download from MS.

Comment (wmhussain) [reply two-tier-app-migration-to-containers-on-Azure/scripts/DotNetAppSqlDb.deploy.cmd]: Application source is https://docs.microsoft.com/en-us/azure/app-service/app-service-web-tutorial-dotnet-sqldatabase and is complied in Visual Studio.

File: two-tier-app-migration-to-containers-on-Azure/scripts/DotNetAppSqlDb.SetParameters.xml (hunk 15 above)
Comment (wmhussain) [reply two-tier-app-migration-to-containers-on-Azure/scripts/DotNetAppSqlDb.SetParameters.xml]: Changed...

File: two-tier-app-migration-to-containers-on-Azure/README.md (hunk 1 above)
Comment (bmoore-msft) [reply two-tier-app-migration-to-containers-on-Azure/README.md]: It should be hosted outside of the repo

File: two-tier-app-migration-to-containers-on-Azure/nested/iis.json [hunk 20]
```diff
@@ -0,0 +1,93 @@
...
+         "metadata": {
+             "description": "Password for the database user dbuser."
+                }
+       },
+    "_artifactsLocation": {
+      "type": "string",
+      "metadata": {
+        "description": "The base URI where artifacts required by this template are located. When the template is deployed using the accompanying scripts, a private location in the subscription will be used and this value will be automatically generated."
+      },
+      "defaultValue": "https://raw.githubusercontent.com/SpektraSystems/2-Tier-app-migration-to-containers-on-Azure/master/"
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/nested/iis.json]: needs to be Azure/master

File: two-tier-app-migration-to-containers-on-Azure/nested/iis.json [hunk 21]
```diff
@@ -0,0 +1,93 @@
...
+      },
+      "defaultValue": ""
+    }
+  },
+  "variables": {
+    "vmExtensionName": "WebDeploy",
+    "nestedCustomTemplateFileName": "webDeploy.json",
+    "DSCFolder": "DSC/",
+    "nestedFolder": "nested/",
+    "dscExtensionFileName": "ContosoWebsite.zip",
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/nested/iis.json]: still need to include the source PS1 files

File: two-tier-app-migration-to-containers-on-Azure/nested/webDeploy.json [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+    "scriptFolder": "scripts/",
+    "scriptFileName": "envcreate.ps1",
+    "dbcreateScript": "[concat(parameters('_artifactsLocation'), variables('scriptFolder'), variables('scriptFileName'), parameters('_artifactsLocationSasToken'))]",
+    "url1": "[concat(parameters('_artifactsLocation'), 'scripts/DotNetAppSqlDb.deploy.cmd', parameters('_artifactsLocationSasToken'))]",
+    "url2": "[concat(parameters('_artifactsLocation'), 'scripts/DotNetAppSqlDb.SetParameters.xml', parameters('_artifactsLocationSasToken'))]",
+    "url3": "[concat(parameters('_artifactsLocation'), 'scripts/DotNetAppSqlDb.zip', parameters('_artifactsLocationSasToken'))]",
+  },
+  "resources": [
+    {
+      "apiVersion": "2015-06-15",
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/nested/webDeploy.json]: update apiVersion

File: two-tier-app-migration-to-containers-on-Azure/nested/webDeploy.json [hunk 23]
```diff
@@ -0,0 +1,64 @@
...
+      "defaultValue": ""
+    }
+  },
+  "variables": {
+    "scriptFolder": "scripts/",
+    "scriptFileName": "envcreate.ps1",
+    "dbcreateScript": "[concat(parameters('_artifactsLocation'), variables('scriptFolder'), variables('scriptFileName'), parameters('_artifactsLocationSasToken'))]",
+    "url1": "[concat(parameters('_artifactsLocation'), 'scripts/DotNetAppSqlDb.deploy.cmd', parameters('_artifactsLocationSasToken'))]",
+    "url2": "[concat(parameters('_artifactsLocation'), 'scripts/DotNetAppSqlDb.SetParameters.xml', parameters('_artifactsLocationSasToken'))]",
+    "url3": "[concat(parameters('_artifactsLocation'), 'scripts/DotNetAppSqlDb.zip', parameters('_artifactsLocationSasToken'))]",
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/nested/webDeploy.json]: this needs to come from a secure location, we can't check compiled binaries into the sample (they can't be verified)

File: two-tier-app-migration-to-containers-on-Azure/nested/webDeploy.json [hunk 24]
```diff
@@ -0,0 +1,64 @@
...
+         "metadata": {
+             "description": "Password for the database user dbuser."
+                }
+       },
+    "_artifactsLocation": {
+      "type": "string",
+      "metadata": {
+        "description": "The base URI where artifacts required by this template are located. When the template is deployed using the accompanying scripts, a private location in the subscription will be used and this value will be automatically generated."
+      },
+      "defaultValue": "https://raw.githubusercontent.com/SpektraSystems/two-tier-app-migration-containers/master/"
```
Comment (bmoore-msft) [inline two-tier-app-migration-to-containers-on-Azure/nested/webDeploy.json]: needs to be Azure/master

File: two-tier-app-migration-to-containers-on-Azure/README.md (hunk 1 above)
Comment (wmhussain) [reply two-tier-app-migration-to-containers-on-Azure/README.md]: deleted the pdf file, link is now pointing to pdf file in our github repo.

File: two-tier-app-migration-to-containers-on-Azure/nested/iis.json (hunk 20 above)
Comment (wmhussain) [reply two-tier-app-migration-to-containers-on-Azure/nested/iis.json]: done

File: two-tier-app-migration-to-containers-on-Azure/nested/webDeploy.json (hunk 24 above)
Comment (wmhussain) [reply two-tier-app-migration-to-containers-on-Azure/nested/webDeploy.json]: done

File: two-tier-app-migration-to-containers-on-Azure/nested/iis.json (hunk 21 above)
Comment (wmhussain) [reply two-tier-app-migration-to-containers-on-Azure/nested/iis.json]: I have copied this DSC extension from [here](https://github.com/Azure/azure-quickstart-templates/tree/master/dsc-extension-iis-server-windows-vm) to deploy IIS on top of SQL VM. 
I am not able to understand what do you mean by Source PS1 files.

Comment (bmoore-msft) [reply two-tier-app-migration-to-containers-on-Azure/nested/iis.json]: Inside that zip file there is a PS1 file.  Place that file into a folder named "dsc" in this sample

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback.

**Pull request closed without merging**